test/TestUtilities/Test.Utility/SimpleTestSetup/NuGetScratchEnvironmentContext.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using NuGet.Common;
+using Xunit;
+
+namespace Test.Utility.SimpleTestSetup
+{
+    public enum NuGetScratchPathKind
+    {
+        Default,
+        Custom
+    }
+
+    public class NuGetScratchEnvironmentContext : IDisposable
+    {
+        private readonly string _backupValue;
+        public NuGetScratchEnvironmentContext(NuGetScratchPathKind nuGetScratchKind)
+        {
+            _backupValue = Environment.GetEnvironmentVariable("NUGET_SCRATCH");
+            if (nuGetScratchKind == NuGetScratchPathKind.Custom)
+            {
+                var expectedNuGetScratchPath = NuGetEnvironment.GetFolderPath(NuGetFolderPath.Temp) + "6b659c0de64e";
+                Environment.SetEnvironmentVariable("NUGET_SCRATCH", expectedNuGetScratchPath);
+                Assert.Equal(expectedNuGetScratchPath, NuGetEnvironment.GetFolderPath(NuGetFolderPath.Temp));
+            }
+        }
+
+        public void Dispose()
+        {
+            Environment.SetEnvironmentVariable("NUGET_SCRATCH", _backupValue);
```

Comment:
How about adding a NuGetEnvironmentVariableWrapper internal static to make the tests not rely on setting actual environment variables.

Reply:
Many of our test assemblies run tests in parallel, and it's unclear to me whether `dotnet test` or xunit run assemblies in parallel (in the same process).

nuget.exe (NuGet.CommandLine.FuncTests) and dotnet.exe tests (dotnet.integration.tests) that run tests by starting a new process, can pass environment variables to the ProcessStartInfo, allowing each test to use a different value, but they're slow because they have to start new processes, wait for JIT compile, run the normal product code start to end, etc. Plus you can only run tests for production scenarios.

Not only is it dangerous to have tests that modify global/process state, since it can introduce flakiness, especially with regards to parallel tests, but I think a helper method in a utility assembly that does this adds a lot of risk to future flaky tests.

Working on NuGet is challenging because it's not only a tool, but also APIs that many people complain about when we break APIs. But if the production code can't be set up to use `IEnvironmentVariableReader`, and therefore be testable with mocks, then I'm not sure it's worth adding automated tests for.

Reply:
The NuGetEnvironment class is static, so an internal static should do it.

Reply:
Statics are global state, so it's not immediately clear to me how that eliminates risks of parallel tests.

Reply:
We can make that test class not run in parallel. 

I'm not too concerned about having a test here, so I'm comfortable with removing the test.

Reply:
So, what is the decision? Do I remove the unit test at all or do I introduce static properties the abstract away environment (and possibly `Path.GetTempPath`) ?

Reply:
Just remove the unit test.